kernel/constr.mli

```diff
@@ -227,7 +243,7 @@ type ('constr, 'types, 'sort, 'univs) kind_of_term =
 
   | Ind       of (inductive * 'univs)                 (** A name of an inductive type defined by [Variant], [Inductive] or [Record] Vernacular-commands. *)
   | Construct of (constructor * 'univs)              (** A constructor of an inductive type defined by [Variant], [Inductive] or [Record] Vernacular-commands. *)
-  | Case      of case_info * 'constr * 'constr * 'constr array
+  | Case      of case_info * 'univs * 'constr array * 'types pcase_return * 'constr * 'constr pcase_branch array
```

Comment:
This may be a good time to consider using a record `Case of { case_info : case_info; ... }` WDYT?

Reply:
Yeah but then the stupid warning for missing matched fields that is active on the kernel is going to be a PITA.

Reply:
There's plenty of empirical evidence that the warning is quite useful. What's the problem in adding `; _`?

Reply:
It's not future-proof. Next time you add a field, the compiler is not going to help you. The alternative solution is to list all the fields but its really cumbersome.

Reply:
This is exactly why I think this warning on by default is a very silly choice.

If you use records, it is exactly because you don't want your code to break when you add a field.
Otherwise use a simple ADT with one constructor.

Reply:
While you are at it, you could use a record `Match of { ... }` instead ;)